php-checks/src/test/resources/checks/regex/InvalidRegexCheck.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace checks\regex;
+
+class InvalidRegexCheck
+{
+  function noncompliant($input)
+  {
+    preg_match('/(/', $input); // Noncompliant {{Fix the syntax error inside this regex.}}
+    preg_match('/x{1,2,3}|(/', $input); // Noncompliant {{Fix the syntax errors inside this regex.}}
+    preg_match('/$[a-z^/', $input); // Noncompliant
+    preg_match("/(\\w+-(\\d+)/", $input); // Noncompliant
```

Comment:
I think it would be great to check that we correctly highlight the precise location and the secondary locations (when there are).